controllers/medusa/medusarestorejob_controller.go

```diff
@@ -262,26 +263,36 @@ func (r *MedusaRestoreJobReconciler) podTemplateSpecUpdateComplete(ctx context.C
 }
 
 // prepareRestore prepares the MedusaRestoreMapping for the restore operation.
-// It uses the Medusa client to get the host map for the restore operation, using the Medusa standalone service to get the backup topology.
+// It uses the Medusa client to get the host map for the restore operation, using the first pod answering on the backup sidecar port.
 func (r *MedusaRestoreJobReconciler) prepareRestore(ctx context.Context, request *medusa.RestoreRequest) (*medusav1alpha1.MedusaRestoreMapping, error) {
-	medusaClient, err := r.ClientFactory.NewClient(ctx, medusaServiceUrl(
-		cassdcapi.CleanupForKubernetes(request.Datacenter.Spec.ClusterName),
-		request.Datacenter.SanitizedName(),
-		request.Datacenter.Namespace))
-	if err != nil {
-		return nil, err
-	}
-	restoreHostMap, err := medusa.GetHostMap(request.Datacenter, *request.RestoreJob, medusaClient, ctx)
+	pods, err := medusa.GetCassandraDatacenterPods(ctx, request.Datacenter, r, request.Log)
 	if err != nil {
+		request.Log.Error(err, "Failed to get datacenter pods")
 		return nil, err
 	}
 
-	medusaRestoreMapping, err := restoreHostMap.ToMedusaRestoreMapping()
-	if err != nil {
-		return nil, err
-	}
+	for _, pod := range pods {
+		addr := net.JoinHostPort(pod.Status.PodIP, fmt.Sprint(shared.BackupSidecarPort))
```

Comment:
Issue: I'm not sure how this is supposed to work, on the one hand the spec in the ticket calls for there to be logic which sequentially tries different pods until it finds one to whose Medusa container it can connect. 

But that isn't what is happening here. If I'm not mistaken, you're going to error out if you can't connect to the first pod you try. Moreover, if the ordering of the pods is deterministic, then every time you hit this loop you'll obtain the same error, and you'll never progress to trying other sidecar containers in other pods which may be available.

If you have to do it this way (rather than having some real retry logic) at least consider randomising the order of the pods in this slice so that you're trying more than just the first one.

Reply:
The first check of `err != nil` in the `else` clause needs to be removed for this to work.
Not sure how it ended up there, good catch 👍

Reply:
Sounds about right. Suggestion: I think you should have a test here and have a mock where the first pod fails and the logic has to move on to another pod. Otherwise we aren't checking that this actually meets the specification.